src/components/error-overlay/error-overlay.js

```diff
@@ -0,0 +1,155 @@
+//@flow
+import style from '../../styles/style.scss';
+import {h} from 'preact';
+import {connect} from 'preact-redux';
+import {default as Icon, IconType} from '../icon';
+import BaseComponent from '../base';
+import Overlay from '../overlay';
+
+/**
+ * mapping state to props
+ * @param {*} state - redux store state
+ * @returns {Object} - mapped state to this component
+ */
+const mapStateToProps = state => ({
+  error: state.engine.error
+});
+
+const defaultError = "Something went wrong";
+const defaultSessionText = "Session ID ";
+const buttonText = "Retry";
+
+@connect(mapStateToProps)
+  /**
+   * errorOverlay component
+   *
+   * @class errorOverlay
+   * @extends {BaseComponent}
+   */
+class ErrorOverlay extends BaseComponent {
+  /**
+   * Creates an instance of ErrorObejct.
+   * @param {Object} obj obj
+   * @memberof ErrorObejct
+   */
+  constructor(obj: any) {
+    super({name: 'ErrorOverlay', player: obj.player});
+  }
+
+  /**
+   * copy input text based on input element.
+   * on success, set success internal component state for 2 seconds
+   * @returns {void}
+   * @memberof ShareOverlay
+   */
+  copyError(): void {
+    let selection = window.getSelection();
+    let dataEl = document.querySelector("." + style.errorSession);
+    let range = document.createRange();
+    try {
+      range.selectNode(dataEl || document.createElement('div'));
+      selection.removeAllRanges();
+      selection.addRange(range);
+      document.execCommand('copy');
+      this.setState({copySuccess: true});
+      setTimeout(() => this.setState({copySuccess: false}), 2000);
+    } catch (e) {
+      this.setState({copySuccess: false});
+    }
+  }
+
+  /**
+   * play on click
+   *
+   * @returns {void}
+   * @memberof PrePlaybackPlayOverlay
+   */
+  handleClick(): void {
+    new Promise((resolve, reject) => {
+      try {
+        if (this.player.config.playback.preload === "auto" && !this.player.config.plugins.ima) {
+          this.player.ready().then(resolve);
+        } else {
+          resolve();
+        }
+      } catch (e) {
+        reject(e);
+      }
+    }).then(() => {
+      this.player.play();
+    }).catch((e) => {
+      this.logger.error(e.message);
+    });
+  }
+
+  /**
+   * render main state
+   *
+   * @returns {React$Element} - main state element
+   * @memberof ErrorOverlay
+   */
+  render(): React$Element<any> | void {
+    let copyUrlClasses = [style.btnCopyUrl].join(' ');
```

Comment:
In general think on split things down here to ui components for reuse and readability.
In addition, I think that too much definitions are inline here - can we export it to css files?
What about ids to dom elements? do we really need it?